-code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-48046/44766

ERROR: Build errors found during clang-tidy run.
```
/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02888/el8_amd64_gcc12/cms/cmssw-patch/CMSSW_15_1_X_2025-05-08-2300/src/DataFormats/ProtonReco/interface/ForwardProtonFwd.h:20:9: note: forward declaration of 'reco::ForwardProton'
--
src/Validation/CTPPS/plugins/CTPPSProtonReconstructionPlotter.cc:249:12: error: member access into incomplete type 'const reco::ForwardProton' [clang-diagnostic-error]
  249 |       if (p.ndof() > 0)
      |            ^
/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02888/el8_amd64_gcc12/cms/cmssw-patch/CMSSW_15_1_X_2025-05-08-2300/src/DataFormats/ProtonReco/interface/ForwardProtonFwd.h:20:9: note: forward declaration of 'reco::ForwardProton'
--
src/Validation/CTPPS/plugins/CTPPSProtonReconstructionPlotter.cc:250:30: error: member access into incomplete type 'const reco::ForwardProton' [clang-diagnostic-error]
  250 |         h_chi_sq_norm->Fill(p.normalizedChi2());
      |                              ^
/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02888/el8_amd64_gcc12/cms/cmssw-patch/CMSSW_15_1_X_2025-05-08-2300/src/DataFormats/ProtonReco/interface/ForwardProtonFwd.h:20:9: note: forward declaration of 'reco::ForwardProton'
--
src/Validation/CTPPS/plugins/CTPPSProtonReconstructionPlotter.cc:255:19: error: member access into incomplete type 'const reco::ForwardProton' [clang-diagnostic-error]
  255 |       h_xi->Fill(p.xi());
      |                   ^
/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02888/el8_amd64_gcc12/cms/cmssw-patch/CMSSW_15_1_X_2025-05-08-2300/src/DataFormats/ProtonReco/interface/ForwardProtonFwd.h:20:9: note: forward declaration of 'reco::ForwardProton'
--
src/Validation/CTPPS/plugins/CTPPSProtonReconstructionPlotter.cc:260:22: error: member access into incomplete type 'const reco::ForwardProton' [clang-diagnostic-error]
  260 |       h_vtx_y->Fill(p.vertex().y());
      |                      ^
/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02888/el8_amd64_gcc12/cms/cmssw-patch/CMSSW_15_1_X_2025-05-08-2300/src/DataFormats/ProtonReco/interface/ForwardProtonFwd.h:20:9: note: forward declaration of 'reco::ForwardProton'
--
gmake: *** [config/SCRAM/GMake/Makefile.coderules:129: code-checks] Error 2
gmake: *** [There are compilation/build errors. Please see the detail log above.] Error 2
```

--- a/Validation/CTPPS/plugins/CTPPSProtonReconstructionPlotter.cc
+++ b/Validation/CTPPS/plugins/CTPPSProtonReconstructionPlotter.cc
@@ -701,6 +701,9 @@ void CTPPSProtonReconstructionPlotter::analyze(const edm::Event &event, const ed
     if (!proton.validFit())
       continue;
 
+    if (proton.contributingLocalTracks().begin() == proton.contributingLocalTracks().end()) {
+      continue;
+    }
     CTPPSDetId rpId((*proton.contributingLocalTracks().begin())->rpId());
     unsigned int armId = rpId.arm();
     const auto &pl = multiRPPlots_[armId];

